shared/src/business/useCases/serveCaseToIrs/serveCaseToIrsInteractor.js

```diff
@@ -151,32 +153,51 @@ const generateNoticeOfReceipt = async ({ applicationContext, caseEntity }) => {
       },
     });
 
-  const contactSecondary = caseEntity.petitioners[1];
-  if (contactSecondary) {
-    const dataWithNoticeAttached = await generatePaperNoticeForContactSecondary(
-      {
+  let secondaryContactNotrPdfData;
+  const contactSecondary = caseEntity.getContactSecondary();
+  const addressesAreDifferent = contactAddressesAreDifferent({
+    applicationContext,
+    caseEntity,
+  });
+  if (contactSecondary && addressesAreDifferent) {
+    secondaryContactNotrPdfData = await applicationContext
+      .getDocumentGenerators()
+      .noticeOfReceiptOfPetition({
         applicationContext,
-        caseCaptionExtension,
-        caseEntity,
-        caseTitle,
-        contactSecondary,
-        docketNumberWithSuffix,
-        pdfData,
-        preferredTrialCity,
-        receivedAt,
-      },
-    );
-    if (dataWithNoticeAttached) {
-      pdfData = dataWithNoticeAttached;
-    }
+        data: {
+          address: contactSecondary,
+          caseCaptionExtension,
+          caseTitle,
+          docketNumberWithSuffix,
+          preferredTrialCity,
+          receivedAtFormatted: applicationContext
+            .getUtilities()
+            .formatDateString(receivedAt, 'MONTH_DAY_YEAR'),
+          servedDate: applicationContext
+            .getUtilities()
+            .formatDateString(caseEntity.getIrsSendDate(), 'MONTH_DAY_YEAR'),
+        },
+      });
   }
 
-  const caseConfirmationPdfName =
-    caseEntity.getCaseConfirmationGeneratedPdfFileName();
-
-  const isProSe = caseEntity.privatePractitioners.length === 0;
-
-  if (preferredTrialCity && isProSe) {
+  let clinicLetter;
+  const isPrimaryContactProSe =
```

Comment:
There is also a method off of the Case entity (isUserIdRepresentedByPrivatePractitioner) that just checks if the user is represented, that may be helpful here?